util/http/http_handler.cc

```diff
@@ -357,13 +362,18 @@ void HttpConnection::HandleRequests() {
 
   while (true) {
     ParserType parser;
-    parser.eager(true);
 
+    // h2::read sets parser.eager=true
+    // This function reads from the socket until parser reaches done state or error is encountered.
     h2::read(asa, req_buffer_, parser, ec);
     if (ec) {
       break;
     }
 
+    if (!parser.is_done()) {
+      LOG(DFATAL) << "Incomplete http parsing";
```

Comment:
I'd expect to see a log level not more than warning, as this might mean that the client is misbehaving or that the connection was dropped?

Reply:
Look at the comment above. h2::read should not allow this

Reply:
Oh, so the above `if (ec) { ... }` handles a bad request? gotcha.